src/pages/feedback.tsx

```diff
@@ -0,0 +1,232 @@
+import AuthenticationServerSidePropsContext from "types/AuthenticationServerSidePropsContext"
+import { isError } from "types/Result"
+import { SurveyFeedbackResponse, SurveyFeedbackType } from "types/SurveyFeedback"
+import { isPost } from "utils/http"
+import parseFormData from "utils/parseFormData"
+import redirectTo from "utils/redirectTo"
+
+enum FeedbackExperienceKey {
+  verySatisfied,
+  satisfied,
+  neutral,
+  dissatisfied,
+  veryDissatisfied
+}
+
+const FeedbackExperienceOptions: KeyValuePair<FeedbackExperienceKey, string> = {
+  0: "Very satisfied",
+  1: "Satisfied",
+  2: "Neither satisfied nor dissatisfied",
+  3: "Dissatisfied",
+  4: "Very dissatisfied"
+}
+
+export const getServerSideProps = withMultipleServerSideProps(
+  withAuthentication,
+  async (context: GetServerSidePropsContext<ParsedUrlQuery>): Promise<GetServerSidePropsResult<Props>> => {
+    const { currentUser, query, req } = context as AuthenticationServerSidePropsContext
+    const { previousPath } = query as { previousPath: string }
+
+    const dataSource = await getDataSource()
+
+    const props = {
+      user: currentUser.serialize(),
+      previousPath,
+      displayAnonymousMissingError: false,
+      displayExperienceMissingError: false,
+      displayfeedbackMissingError: false
+    }
+
+    if (isPost(req)) {
+      const { isAnonymous, experience, feedback } = (await parseFormData(req)) as {
+        isAnonymous: string
+        experience: string
+        feedback: string
+      }
+
+      if (isAnonymous && experience && feedback) {
+        const result = await insertSurveyFeedback(dataSource, {
+          feedbackType: SurveyFeedbackType.General,
+          userId: isAnonymous === "no" ? currentUser.id : null,
+          response: { isAnonymous, experience, comment: feedback } as SurveyFeedbackResponse
+        } as SurveyFeedback)
+
+        if (!isError(result)) {
+          return redirectTo(previousPath)
+        } else {
+          throw result
+        }
+      }
+
+      return {
+        props: {
+          ...props,
+          displayAnonymousMissingError: !isAnonymous ? true : false,
+          displayExperienceMissingError: !experience ? true : false,
+          displayfeedbackMissingError: !feedback ? true : false,
+          selectedFields: { isAnonymous: isAnonymous ?? null, experience: experience ?? null, feedback }
+        }
+      }
+    }
+
+    return { props }
+  }
+)
+
+interface Props {
+  user: User
+  previousPath: string
+  displayAnonymousMissingError: boolean
+  displayExperienceMissingError: boolean
+  displayfeedbackMissingError: boolean
+  selectedFields?: { isAnonymous: string; experience: string; feedback: string }
+}
+
+const FeedbackPage: NextPage<Props> = ({
+  user,
+  previousPath,
+  displayAnonymousMissingError,
+  displayExperienceMissingError,
+  displayfeedbackMissingError,
+  selectedFields
+}: Props) => {
+  const maxFeedbackNoteLength: number = 2000
```

Comment:
We could set the max feedback length in the config just as we do with note length